code/CodeTester.cpp

```diff
@@ -933,16 +933,115 @@ void rtps_api_example_create_entities()
     }
 
     //RTPS_API_WRITE_SAMPLE
-    //Request a change from the history
+    //Request a change from the writer
     CacheChange_t* change = writer->new_change([]() -> uint32_t {
         return 255;
     }, ALIVE);
     //Write serialized data into the change
     change->serializedPayload.length = sprintf((char*) change->serializedPayload.data, "My example string %d", 2) + 1;
-    //Insert change back into the history. The Writer takes care of the rest.
+    //Insert change into the history. The Writer takes care of the rest.
     history->add_change(change);
     //!--
+}
+
+
+void rtps_api_example_create_entities_with_custom_pool()
+{
+    RTPSParticipantAttributes participant_attr;
+    participant_attr.setName("participant");
+    RTPSParticipant* participant = RTPSDomain::createParticipant(0, participant_attr);
+
+    //RTPS_API_ENTITY_CREATE_WITH_PAYLOAD_POOL
+    // A simple payload pool that reserves and frees memory each time
+    class CustomPayloadPool : public IPayloadPool
+    {
+        bool get_payload(
+                uint32_t size,
+                CacheChange_t& cache_change) override
+        {
+            // Reserve new memory for the payload buffer
+            octet* payload = new octet[size];
+
+            // Assign the payload buffer to the CacheChange and update sizes
+            cache_change.serializedPayload.data = payload;
+            cache_change.serializedPayload.length = size;
+            cache_change.serializedPayload.max_size = size;
+
+            // Tell the CacheChange who needs to release its payload
+            cache_change.payload_owner(this);
+            
+            return true;
+        }
+
+        bool get_payload(
+                SerializedPayload_t& data,
+                IPayloadPool*& /* data_owner */,
+                CacheChange_t& cache_change) override
+        {
+            // Reserve new memory for the payload buffer
+            octet* payload = new octet[data.length];
+
+            // Copy the data
+            memcpy(payload, data.data, data.length);
+
+            // Assign the payload buffer to the CacheChange and update sizes
+            cache_change.serializedPayload.data = payload;
+            cache_change.serializedPayload.length = data.length;
+            cache_change.serializedPayload.max_size = data.length;
+
+            // Tell the CacheChange who needs to release its payload
+            cache_change.payload_owner(this);
+            
+            return true;
+        }
+
+        bool release_payload(
+                CacheChange_t& cache_change) override
+        {
+            // ensure precondition
+            assert (this == cache_change.payload_owner());
```

Comment:
Precondition only check in debug, I guess that's alright for the example
```suggestion
            assert(this == cache_change.payload_owner());
```